lib/camunda-platform/features/modeling/behavior/UpdateCamundaExclusiveBehavior.js

```diff
@@ -0,0 +1,98 @@
+import inherits from 'inherits';
+
+import {
+  getBusinessObject,
+  is
+} from 'bpmn-js/lib/util/ModelUtil';
+
+import CommandInterceptor from 'diagram-js/lib/command/CommandInterceptor';
+
+const HIGH_PRIORITY = 15000;
+
+
+/**
+ * Camunda BPMN specific `camunda:exclusive` behavior.
+ *
+ * When `camunda:asyncAfter` or `camunda:asyncBefore` are set to false
+ * on an element, then `camunda:exclusive` shall be set to true. This ensures
+ * that the BPMN diagram XML reflects the behavior of Camunda Platform engine.
+ */
+export default function UpdateCamundaExclusiveBehavior(eventBus) {
+
+  CommandInterceptor.call(this, eventBus);
+
+  this.preExecute([ 'properties-panel.update-businessobject', 'element.updateProperties' ],
+    HIGH_PRIORITY, function(context) {
+      const {
+        element,
+        properties
+      } = context;
+
+      const businessObject = getBusinessObject(element);
+
+      // (1.1) Execute if...
+      if (is(element, 'camunda:AsyncCapable') && // ...asyncCapable
+      properties && // ...properties updated
+      (properties['camunda:asyncBefore'] === false || // ...update async (1)
+        properties['camunda:asyncAfter'] === false) && // ...update async (2)
+      isExclusive(businessObject) === false // ...is currently not exclusive
+      ) {
+
+        // (1.2) ... but don't execute if both async were set or one is set to true
+        if ((isAsyncAfter(businessObject) && isAsyncBefore(businessObject)) ||
```

Comment:
So it wouldn't execute when `asyncBefore=true` & `asyncAfter=true` _before_ the update? Can't it be the case when I update both properties in one command?

```js
{
  ...
  properties: {
    'camunda:asyncBefore': true,
    'camunda:asyncAfter': true
  }
}
```

That's surely not possible via properties panel, but maybe via manual command / `modeling.updateProperties`.

Reply:
Ah true, I forgot that case. 

I added a test-case and adjusted the logic

Reply:
ty